### ISSUE TITLE
[v2] Admin order editing support

### ISSUE THREAD

**Issue by helgatheviking** [Backcourt/wc-mnm-variable#42]
In the metaboxes, this does not correctly update the order.

1. The modal opening script will need a pure JS event dispatched alongside the [jQuery trigger](https://github.com/kathyisawesome/woocommerce-mix-and-match-products/blob/2.5.0/assets/js/admin/meta-boxes-order.js#L95).

Something like this does work to display the form, though unsure if that's enough variables in the detail:

```
const initEvent = new CustomEvent( `wc-mnm-initialize.${type}`, {
	detail: {
		source: 'metabox',
		target: this,
        },
});
document.dispatchEvent(initEvent);
```

2. It fails to save properly since the ajax is posted the [config that it gets from the default MNM add to cart script](https://github.com/kathyisawesome/woocommerce-mix-and-match-products/blob/2.5.0/assets/js/admin/meta-boxes-order.js#L132). In fact, now that the add to cart script is not enqueued at all for the reactified version... `Form.api` throws an error since `Form` returns `false` instead of undefined. 

Two options for saving...
A. duplicate a lot of the meta box orders script and have separate scripts for variable MNM.
B. Stash the config as a data attribute on a DOM element... like the add to cart button? (which might already be [happening on the simple MNM side]). Then if we hack our way to doing the same on the Variable MNM side and still let the MNM order script handle both.

3. test if editing correctly pre-fills with chosen selections.
4. test you can save the variation and switch variations.

**Comment by helgatheviking** [comment]
Maybe core MNM could use 
```
// Get the form
let form = document.querySelector('#post');

// Get all field data from the form
// returns a FormData object
let data = new FormData(form);
```

To get the form data? The form data exists in the DOM in both traditional MNM and reactified Variable MNM so that could be used on both?

**Comment by helgatheviking** [comment]
Data saves thanks to https://github.com/kathyisawesome/wc-mnm-variable/commit/7c46f3966bb3fcd8747ea7ac484d5c1f3567e9fd now we just need to pre-fill with the existing config. We already support that via URL params, but the config state might need to be set some other way since this is an ajax loaded modal and the URL won't have the values.

**Comment by helgatheviking** [comment]
Closed by 7c46f39 and fad2726eaf19c99806c53fc72c5d51af1899e6c4
